### ISSUE TITLE
Does not recognize computed exception tuples

### ISSUE THREAD

**Issue by helmutg** [python/mypy#21092]
**Bug Report**

mypy does not recognize computed exception tuples.

**To Reproduce**

```python
exceptions: tuple[type[BaseException], ...] = (ZeroDivisionError,)

try:
    1 / 0
except (ValueError,) + exceptions:
    pass
```

**Expected Behavior**

I expected this to show no diagnostics.

**Actual Behavior**

```
main.py:5: error: Exception type must be derived from BaseException (or be a tuple of exception classes)  [misc]
```

**Workaround**

Adding a variable `exceptions_to_catch: tuple[type[BaseException], ...] = (ValueError,) + exceptions` can be used to mitigate the problem.

**Your Environment**

- Mypy version used: 1.19.1-2 (Debian)
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: 3.13.12 (Debian)

The problem can also be reproduced on mypy-play.net

**Real world use case**

When a library is optionally imported (e.g. one of several implementations), the exception type cannot be known in advance. The exception tuple can be used to avoid implementation-dependent code paths. For instance, using `compression.zstd` vs `zstandard` could be achieved with this technique.

**Comment by dhanusharer** [comment]
Hi! I’d like to work on this issue. It seems related to tuple type inference during concatenation. Please let me know if it’s okay to proceed or if there’s any prior discussion I should be aware of.

**Comment by hauntsaninja** [comment]
Thanks, this is fixed in mypy 1.20
